### ISSUE TITLE
Synchronization project files workspace -> Artik device

### ISSUE THREAD

**Issue by vparfonov** [codenvy/artik-ide#89]
### Description
It should be possible to automatically sync project files and binaries with the device, i.e. all changes and new files in `/projects` directory in a workspace are auto synced into the selected directory of a selected device.

When a device is added, it should be possible to select a directory to push files to (`/root` by default)

We should use file watchers to catch changes and an existing `scp` service to push files. Everything should happen in a background without notifications.

### Related Issues:
- [ ] Investigate use of rsync instead of `scp` to replicate project files on the Artik device #90

**Comment by dkuleshov** [comment]
During this work I found that `SCP` has a few limitations:
- It does not recursively create parents if they does not exist. That means that if we are trying to `SCP` file` /root/folder/file` when `/root/folder` does not exist we receive error.
- It opens SSH connection each time we run `SCP`. That rises an optimization problem:
 - If we `SCP` the whole project folder it is redundant if we need to update one file
 - If we `SCP` each file separately it is redundant if we need to update a group of files

Besides that, there are also complications on `VFS` event handling side. The matter is that at the moment of this writing the event handling subsystem is designed to process events in real time. That means that if artik machines are not connected when event happens (one of normal use cases: we create project, then we connect a machine) the event is handled and we understand that the content is modified but we have no machine to replicate it to, so it is not postponed but skipped. In other words right now we can properly replicate content that is modified only after artik machine is connected.
Both of this has impact on proper data replication process for artik related projects, what we can do here is:
- We can try to find replacement for `SCP` (e.g. `rsync`)
- We can implement some service that will complement `SCP` and properly treat skipped events

**Comment by TylerJewell** [comment]
Given the standardization of rsync it seems that would need to be the first option.  Implementing an extension to scp would not be maintainable and I would -1 that solution.

**Comment by dkuleshov** [comment]
Agreed, I will check if `rsync` fits our needs and report back. For now it seems to be a solution for all our problems here at once.
